sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -435,20 +437,23 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
       case StringType if right.foldable =>
         val sdf = classOf[SimpleDateFormat].getName
         val fString = if (constFormat == null) null else constFormat.toString
-        val formatter = ctx.freshName("formatter")
         if (fString == null) {
           ev.copy(code = s"""
             boolean ${ev.isNull} = true;
             ${ctx.javaType(dataType)} ${ev.value} = ${ctx.defaultValue(dataType)};""")
         } else {
+          val formatter = ctx.freshName("formatter")
+          ctx.addMutableState(sdf, formatter, s"""$formatter = null;""")
```

Comment:
Not very familiar with codegen, but I wonder if we can add the instantiation here and avoid the null checking below.
ctx.addMutableState(sdf, formatter, s"""$formatter = new $sdf("$fString");""")